tests/phpunit/Process/Runner/TestRunConstraintCheckerTest.php

```diff
@@ -202,4 +202,85 @@ public function test_run_passes_on_exactly_covered_min_msi(): void
 
         $this->assertTrue($constraintChecker->hasTestRunPassedConstraints());
     }
+
+    /** @dataProvider is_msi_over_minimum_msi_provider */
+    public function test_is_msi_over_minimum_msi(?float $actualMsi, ?float $actualCoveredMsi, float $minMsi, float $minCoveredMsi, string $type, bool $expected): void
+    {
+        $metrics = $this->createMock(MetricsCalculator::class);
+
+        if ($actualMsi !== null) {
```

Comment:
I'm not a fan of `if` statements within a tets, maybe we can split up the test cases?